### PR TITLE
SIMSBIOHUB-572 Service/Controller/Repository Refactor - Telemetry

### PR DESCRIPTION
**Pull request by MacQSL**

### CODE REVIEW

File: bctw-api/src/database/pg.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import { Knex, knex } from 'knex';
```
[the remark this answers is not in the document]
Comment (MacQSL) [reply bctw-api/src/database/pg.ts:1]: Whoops

**Comment by NickPhura** [comment]
Created a new animal in SIMS.
Attached a random device ID (30467)
Chose deployment dates in the past, so they wouldn't collide.
Saved. No issues so far.

Tried to delete the deployment.
Error, in SIMS api logs:
```
{
  timestamp: '2024-05-03 09:29:24',
  level: 'debug',
  logger: 'app',
  message: 'DELETE /api/project/1/survey/1/critters/NaN/deployments/0c66052e-85ad-446a-8d5b-62d18fa3baf5'
}
{
  timestamp: '2024-05-03 09:29:24',
  level: 'error',
  logger: 'app',
  label: 'errorTransformer',
  message: 'ajvError',
  ajvError: {
    instancePath: '/critterId',
    schemaPath: '#/properties/critterId/type',
    keyword: 'type',
    params: { type: 'number' },
    message: 'must be number',
    location: 'path'
  }
}
```

**Comment by NickPhura** [comment]
UI nit:
2 of the controls are missing the param that fills in the border (when the field name hasn't moved to the top). `Device Manufacturer` and `Units` in this case.
image

**Comment by NickPhura** [comment]
When I click the `Add Record` button to add manual telemetry, I get a console error:

```
Warning: `value` prop on `input` should not be null. Consider using an empty string to clear the component or `undefined` for uncontrolled components.
input
./node_modules/@emotion/react/dist/emotion-element-43c6fea0.browser.esm.js/withEmotionCache/<@http://localhost:7107/static/js/bundle.js:78710:66
div
./node_modules/@emotion/react/dist/emotion-element-43c6fea0.browser.esm.js/withEmotionCache/<@http://localhost:7107/static/js/bundle.js:78710:66
InputBase@http://localhost:7107/static/js/bundle.js:112366:83
OutlinedInput@http://localhost:7107/static/js/bundle.js:117626:83
div
./node_modules/@emotion/react/dist/emotion-element-43c6fea0.browser.esm.js/withEmotionCache/<@http://localhost:7107/static/js/bundle.js:78710:66
FormControl@http://localhost:7107/static/js/bundle.js:109016:82
./node_modules/@emotion/react/dist/emotion-element-43c6fea0.browser.esm.js/withEmotionCache/<@http://localhost:7107/static/js/bundle.js:78710:66
TextField@http://localhost:7107/static/js/bundle.js:124010:83
TextFieldDataGrid@http://localhost:7107/static/js/bundle.js:3367:27
div
./node_modules/@mui/x-data-grid/components/cell/GridCell.js/GridCellV7<@http://localhost:7107/static/js/bundle.js:139785:9
div
GridRow@http://localhost:7107/static/js/bundle.js:138103:9
div
```

**Comment by NickPhura** [comment]
Tried adding a manual telemetry.
Clicked save.
The spinner starts, and some of the buttons grey out.
But no console logs or network requests are kicked off. It just hangs.

image

**Pull request merged**